config/src/network_id.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+pub struct NetworkInfo {
+    name: String,
+}
+
+/// A representation of the network being used in communication.
+/// There should only be one of each NetworkId used for a single node, and handshakes should verify
+/// that the NetworkId being used is the same during a handshake, to effectively ensure communication
+/// is restricted to a network.  Network should be checked that it is not the `DEFAULT_NETWORK`
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub enum NetworkId {
+    Validator,
+    Public,
+    Private(NetworkInfo),
+}
+
+const INVALID: &str = "Invalid";
```

Comment:
why?

Reply:
I don't think this necessarily has to be a constant.

The purpose of this is when deserializing, it seemed I needed a default value (probably for the struct that I had before).  So, I added this to ensure that if it had been loaded with a lack of a configuration for a network, it wouldn't just all of a sudden connect to one of the three networks.